### PR TITLE
Include file name in Submissions API response

### LINKED ISSUE #1119: Include filename in the /submissions endpoint
The filename for a submission is ingested via the upload's Content Disposition and is being saved and used for the summary. See [here](https://github.com/cfpb/hmda-platform/blob/e682f7d6c95b870beb6d084cb17ccf2d9e6a6be1/persistence/src/main/scala/hmda/persistence/processing/HmdaRawFile.scala) and [here](https://github.com/cfpb/hmda-platform/blob/88afb4962acdc642eac9e14995706456511a4821/api/src/test/scala/hmda/api/http/institutions/submissions/SubmissionSummaryPathsSpec.scala).

We display the filename at different times, across refreshes, to different users of the same institution, and different filings. Using localStorage doesn't cover all these use cases and disrupts our state modeling.

If the filename is available as a key in the [submission](https://github.com/cfpb/hmda-platform/blob/master/Documents/API.md#latest-submission), this would clean up some nasty code on the front-end and allow us to display the correct filename in all cases for a given submission.

### PR DESCRIPTION
**Pull request by schbetsy**
Closes #1119

### CODE REVIEW

File: persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala [hunk 1]
```diff
@@ -40,6 +41,12 @@ object SubmissionPersistence {
           }
 
           SubmissionState(submissions.updated(i, updatedSub))
+
+        case SubmissionFileNameAdded(id, name) =>
```
Comment (nickgrippin) [inline persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala]: We already have a `name` value defined in this scope, maybe change this to something else?

File: persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala [hunk 2]
```diff
@@ -83,6 +90,17 @@ class SubmissionPersistence(institutionId: String, period: String) extends HmdaP
         sender() ! None
       }
 
+    case AddSubmissionFileName(id, name) =>
```
Comment (nickgrippin) [inline persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala]: Same comment here.

File: persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala (hunk 1 above)
Comment (schbetsy) [reply persistence/src/main/scala/hmda/persistence/institutions/SubmissionPersistence.scala]: Good call. Fixed in e2965f7.

**Comment by schbetsy** [comment]
Woohoo! Travis is finally passing. Ready to merge?

**Comment by nickgrippin** [comment]
I'm getting an unused import warning for `SubmissionSummaryPathsSpec` with the most recent change.

**Pull request merged**